File: ios/ExpoBonjourModule.types.swift
//
//  ExpoBonjourModule.types.swift
//  ExpoBonjour
//
//  Created by Aron on 2025/10/28.
//

import ExpoModulesCore

struct PublishingOptions: Record {
    @Field var name: String = ""
    @Field var port: UInt16 = 80
    @Field var service: String = ""
    @Field var domain: String?
    @Field var txtRecord: [String: String]?
}

File: android/src/main/java/expo/modules/bonjour/ExpoBonjourModule.types.kt
package expo.modules.bonjour

import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record

class PublishingOptions: Record {
    @Field var name: String = ""
    @Field var port: Int = 80
    @Field var service: String = ""
    @Field var domain: String? = null
    @Field var txtRecord: Map<String, String>? = null
}
